src/aura/BGE_TemplateDetails/BGE_TemplateDetailsHelper.js

```diff
@@ -580,13 +580,28 @@
              * @return void.
              ************************************************************/
             function updateSelectedToActive() {
+
+                var activeFieldCountBySObject = {};
+                var activeFieldsBySObject = getActivesBySObject();
+
+                Object.keys(activeFieldsBySObject).forEach(function(sObjectName) {
+                    activeFieldCountBySObject[sObjectName] = activeFieldsBySObject[sObjectName].length;
+                });
+
+                console.log(activeFieldCountBySObject);
+
                 _allFields.forEach(function(currentField) {
                     if (!currentField.isActive && currentField.selected) {
                         currentField.isActive = true;
                         currentField.selected = false;
-                        // TODO: this seems expensive. is there a better way?
-                        var activeFieldsBySObject = getActivesBySObject();
```

Comment:
@randi274 can we revert this back? I think the new performance improvement code is causing an issue.